source/sigv4.c

```diff
@@ -1162,110 +1621,1311 @@ static SigV4Status_t getCredentialScope( SigV4Parameters_t * pSigV4Params,
+        returnStatus = SigV4InvalidParameter;
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static int32_t completeHash( const char * pInput,
+                             size_t inputLen,
+                             char * pOutput,
+                             size_t outputLen,
+                             const SigV4CryptoInterface_t * pCryptoInterface )
+{
+    int32_t hashStatus = -1;
+
+    assert( pOutput != NULL );
+    assert( outputLen > 0 );
+    assert( pCryptoInterface != NULL );
+    assert( pCryptoInterface->hashInit != NULL );
+    assert( pCryptoInterface->hashUpdate != NULL );
+    assert( pCryptoInterface->hashFinal != NULL );
+
+    hashStatus = pCryptoInterface->hashInit( pCryptoInterface->pHashContext );
+
+    if( hashStatus == 0 )
+    {
+        hashStatus = pCryptoInterface->hashUpdate( pCryptoInterface->pHashContext,
+                                                   pInput, inputLen );
+    }
+
+    if( hashStatus == 0 )
+    {
+        hashStatus = pCryptoInterface->hashFinal( pCryptoInterface->pHashContext,
+                                                  pOutput, outputLen );
+    }
+
+    return hashStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static SigV4Status_t completeHashAndHexEncode( const char * pInput,
+                                               size_t inputLen,
+                                               char * pOutput,
+                                               size_t * pOutputLen,
+                                               const SigV4CryptoInterface_t * pCryptoInterface )
+{
+    SigV4Status_t returnStatus = SigV4Success;
+    /* Used to store the hash of the request payload. */
+    char hashBuffer[ SIGV4_HASH_MAX_DIGEST_LENGTH ];
+    SigV4String_t originalHash;
+    SigV4String_t hexEncodedHash;
+
+    assert( pOutput != NULL );
+    assert( pOutputLen != NULL );
+    assert( pCryptoInterface != NULL );
+    assert( pCryptoInterface->hashInit != NULL );
+    assert( pCryptoInterface->hashUpdate != NULL );
+    assert( pCryptoInterface->hashFinal != NULL );
+
+    originalHash.pData = hashBuffer;
+    originalHash.dataLen = pCryptoInterface->hashDigestLen;
+    hexEncodedHash.pData = pOutput;
+    hexEncodedHash.dataLen = *pOutputLen;
+
+    if( completeHash( pInput,
+                      inputLen,
+                      hashBuffer,
+                      pCryptoInterface->hashDigestLen,
+                      pCryptoInterface ) != 0 )
+    {
+        returnStatus = SigV4HashError;
+    }
+
+    if( returnStatus == SigV4Success )
+    {
+        /* Hex-encode the request payload. */
+        returnStatus = lowercaseHexEncode( &originalHash,
+                                           &hexEncodedHash );
+    }
+
+    if( returnStatus == SigV4Success )
+    {
+        *pOutputLen = hexEncodedHash.dataLen;
+    }
+
+    return returnStatus;
+}
+
```

Comment:
Functions needs to be separated using `/*-----------------------------------------------------------*/`